src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1204,10 +1205,10 @@ private string GetPropertyNameInitialization()
 
                 StringBuilder sb = new();
 
-                foreach (string propName in _currentContext.RuntimePropertyNames)
+                foreach (KeyValuePair<string, string> name_varName_pair in _currentContext.RuntimePropertyNames)
                 {
                     sb.Append($@"
```

Comment:
Should we also emit an XML comment declaring the unencoded form?

Reply:
I'm open on this. We could add the first occurrence easily. Also discussed in the future options in the bottom of https://github.com/dotnet/runtime/issues/57836#issuecomment-904850749.